### PR TITLE
Feature/2969 inform about risk status change to green

### PR DESCRIPTION
**Pull request by nickguendling**
## Description
Shows an alert when the risk level changed from increased to low.

## Story
JIRA: https://jira.itc.sap.com/browse/EXPOSUREAPP-2969

## Screenshot
IMG_F6C09ED125DF-1

### CODE REVIEW

**Review by KaiTeuber — APPROVED** [review]

**Review by janetback — APPROVED** [review]
UA reviewed.

File: src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/ExposureDetectionViewController+DynamicTableViewModel.swift [hunk 1]
```diff
@@ -458,7 +458,7 @@ extension ExposureDetectionViewController {
 		let activeTracing = state.risk?.details.activeTracing ?? .init(interval: 0)
 		return DynamicTableViewModel([
 			riskDataSection(cells: [
-				.riskContacts(text: AppStrings.ExposureDetection.numberOfContacts, image: UIImage(named: "Icons_RisikoBegegnung")),
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/ExposureDetectionViewController+DynamicTableViewModel.swift:461]: This was a cleanup together with removing the `ExposureDetection_NumberOfContacts` from the stringsdict?

File: src/xcode/ENA/ENA/Source/Scenes/Home/HomeViewController.swift [hunk 2]
```diff
@@ -207,6 +207,28 @@ final class HomeViewController: UIViewController, RequiresAppDependencies {
 		homeInteractor.state.risk = risk
 
 		reloadData(animatingDifferences: false)
+
+		showRiskStatusLoweredAlert(currentRisk: risk)
+	}
+
+	func showRiskStatusLoweredAlert(currentRisk: Risk?) {
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Scenes/Home/HomeViewController.swift:214]:
```suggestion
	func showRiskStatusLoweredAlertIfNeeded {
```

File: src/xcode/ENA/ENA/Source/Services/Risk/Provider/RiskProvider.swift [hunk 3]
```diff
@@ -316,8 +316,21 @@ extension RiskProvider: RiskProviding {
 				return
 		}
 
+		/// Only set shouldShowRiskStatusLoweredAlert if risk level has changed from increase to low or vice versa. Otherwise leave shouldShowRiskStatusLoweredAlert unchanged.
+		/// Scenario: Risk level changed from increased to low in the first risk calculation. in a second risk calculation it stays low. If the user does not open the app between these two calculations, the alert should still be shown.
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Services/Risk/Provider/RiskProvider.swift:320]:
```suggestion
		/// Scenario: Risk level changed from increased to low in the first risk calculation. In a second risk calculation it stays low. If the user does not open the app between these two calculations, the alert should still be shown.
```

**Review by marcussc — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/ExposureDetectionViewController+DynamicTableViewModel.swift (hunk 1 above)
Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/ExposureDetectionViewController+DynamicTableViewModel.swift:461]: Yep

File: src/xcode/ENA/ENA/Source/Scenes/Home/HomeViewController.swift (hunk 2 above)
Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Scenes/Home/HomeViewController.swift]: Done

**Review by marcussc — APPROVED** [review]

**Pull request merged**